src/Ethereum/ABI/Array.cpp

```diff
@@ -133,4 +133,44 @@ std::string ParamArray::getExtraTypes(std::vector<std::string>& ignoreList) cons
     return (proto != nullptr) ? proto->getExtraTypes(ignoreList) : "";
 }
 
+void ParamArrayFix::encode(Data& data) const {
+    this->_params.encode(data);
+}
+
+bool ParamArrayFix::decode(const Data& encoded, size_t& offset_inout) {
+    std::size_t origOffset = offset_inout;
+    auto res = this->_params.decode(encoded, offset_inout);
+    // padding
+    offset_inout = origOffset + ValueEncoder::paddedTo32(offset_inout - origOffset);
+    return res;
```

Comment:
```suggestion
    return this->_params.decode(encoded, offset_inout);
```

Padding here does not hurt, but in fact it is not needed. If it is done here, it should check for size, otherwise it could happen that offset runs out of input data. Better to leave it out completely. It is also difficult to test.

Reply:
done